src/content/datachannel/datatransfer/js/test.js

```diff
@@ -1,20 +1,57 @@
 /*
- *  Copyright (c) 2018 The WebRTC project authors. All Rights Reserved.
+ *  Copyright (c) 2015 The WebRTC project authors. All Rights Reserved.
  *
  *  Use of this source code is governed by a BSD-style license
  *  that can be found in the LICENSE file in the root of the source
  *  tree.
  */
-export default {
-  'It should transfer data over data channel': (browser) => {
-    const path = '/src/content/datachannel/datatransfer/index.html';
-    const url = 'file://' + process.cwd() + path;
-
-    browser
-      .url(url)
-      .click('#sendTheData')
-      .pause(1000)
-      .assert.value('#receiveProgress', '16777200')
-      .end();
-  }
-};
\ No newline at end of file
+/* eslint-env node, mocha */
+
+'use strict';
+const webdriver = require('selenium-webdriver');
+const seleniumHelpers = require('../../../../../test/webdriver');
+const {expect} = require('chai');
+
+let driver;
+const path = '/src/content/datachannel/datatransfer/index.html';
+const url = `${process.env.BASEURL ? process.env.BASEURL : ('file://' + process.cwd())}${path}`;
+
+describe('datachannel datatransfer', () => {
+  before(() => {
+    driver = seleniumHelpers.buildDriver();
+  });
+  after(() => {
+    return driver.quit();
+  });
+
+  beforeEach(() => {
+    return driver.get(url);
+  });
+
+  it('transfers data', async () => {
+    const megsToSend = 4;
+    await driver.findElement(webdriver.By.id('megsToSend'))
+        .clear();
+    await driver.findElement(webdriver.By.id('megsToSend'))
+        .sendKeys(megsToSend + '\n');
+
+    await driver.findElement(webdriver.By.id('sendTheData')).click();
+
+    await Promise.all([
+      driver.wait(() => driver.executeScript(() => {
+        return localConnection && localConnection.connectionState === 'connected'; // eslint-disable-line no-undef
+      })),
+      await driver.wait(() => driver.executeScript(() => {
+        return remoteConnection && remoteConnection.connectionState === 'connected'; // eslint-disable-line no-undef
```

Comment:
that's a lot of no-undef comments. is remoteConnection that hard to see?

Reply:
it is more :exploding_head: 

The tests are linted in the node environment (see the eslint-env at the top) but the executeScript is actually run in the browser. The linter can not know that the variables are defined in the main.js file hence these suppression for no-undef is  needed.

The alternative would be to turn the "in browser" tests into strings (which is how it was done in the past) but those can't be linted at all.

Reply:
I think I've seen comment syntax that declares the variables to Lint.
Second answer in https://stackoverflow.com/questions/41552041/global-variables-in-javascript-and-eslint says tht

/* global remoteConnection */

should be enough to make Lint shut up (and still catch other misspelled variables on that line, if any).

Reply:
[global](https://eslint.org/docs/user-guide/configuring/language-options#specifying-globals) is just that - a global in the node environment which we don't want.

https://eslint.org/docs/rules/no-undef is more close to the use-case. Risk should be minimal since the test needs to work in the end. This should be sufficient to protect us from my butterfingers typing a utf8 character in the wrong place ;-)